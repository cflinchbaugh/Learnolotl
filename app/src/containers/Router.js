import React, { Component } from 'react';
import styled from 'styled-components';

import { BrowserRouter } from 'react-router-dom';
import Route from 'react-router-dom/Route';

import Navigation from '../components/navigation/Navigation';
import LearnContainer from '../components/pages/learn/LearnContainer';
import LearnFlashcardsContainer from '../components/pages/learn/LearnFlashcardsContainer'
import BuildContainer from '../components/pages/build/BuildContainer';
import About from '../components/pages/About';

const StyleWrapper = styled.div`
    max-width: 1400px;
    margin: auto;
    padding: 0 25px;
`;

class Router extends Component {
    render() {
        return (
        <BrowserRouter>
            <div className="Router">
                <Navigation/>
                
                <StyleWrapper>
                    <Route path="/" exact component={LearnContainer}/>

                    <Route path="/learn" exact component={LearnContainer}/>
                    <Route path="/learn/flashcards" component={LearnFlashcardsContainer}/>

                    <Route path="/build" component={BuildContainer}/>
                    <Route path="/about" component={About}/>
                </StyleWrapper>
            </div>
        </BrowserRouter>
        );
    }
}

export default Router;
