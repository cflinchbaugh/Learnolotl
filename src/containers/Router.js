import React, { Component } from 'react';
import styled from 'styled-components';

import { BrowserRouter } from 'react-router-dom';
import Route from 'react-router-dom/Route';

import NavigationContainer from '../components/navigation/NavigationContainer';
import LearnContainer from '../components/pages/learn/LearnContainer';
import LearnFlashcardsContainer from '../components/pages/learn/LearnFlashcardsContainer'
import BuildContainer from '../components/pages/build/BuildContainer';
import About from '../components/pages/About';

const StyleWrapper = styled.div`
    margin: auto;
    padding: 0 25px;
`;

class Router extends Component {
    render() {
        return (
            <BrowserRouter>
                <div className="Router 
                    col-xs-12 col-sm-8 col-sm-offset-2 
                    col-lg-6 col-lg-offset-3
                    clearfix">
                    <NavigationContainer/>
                    
                    <StyleWrapper className="route-content-wrapper col-xs-12">
                        <Route path="/" exact component={LearnContainer}/>
                        <Route path="/Learnolotl/" exact component={LearnContainer}/>

                        <Route path="/Learnolotl/learn" exact component={LearnContainer}/>
                        <Route path="/learn/flashcards" component={LearnFlashcardsContainer}/>

                        <Route path="/Learnolotl/build" component={BuildContainer}/>
                        <Route path="/Learnolotl/about" component={About}/>
                    </StyleWrapper>
                </div>
            </BrowserRouter>
        );
    }
}

export default Router;
