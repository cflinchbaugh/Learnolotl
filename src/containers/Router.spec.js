import React from 'react';
import {shallow} from 'enzyme';
import Router from './Router';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Router', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const routerData = {
            
        }

        const tree = renderer.render(
            <Router {...routerData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    
});