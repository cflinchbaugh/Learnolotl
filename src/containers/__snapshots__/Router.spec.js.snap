// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router renders and matches snapshot 1`] = `
<BrowserRouter>
  <div
    className="Router 
 col-xs-12 col-sm-8 col-sm-offset-2 
 col-lg-6 col-lg-offset-3
 clearfix"
  >
    <Connect(NavigationContainer) />
    <styled.div
      className="route-content-wrapper col-xs-12"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/Learnolotl/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/Learnolotl/learn"
      />
      <Route
        component={[Function]}
        path="/learn/flashcards"
      />
      <Route
        component={[Function]}
        path="/Learnolotl/build"
      />
      <Route
        component={[Function]}
        path="/Learnolotl/about"
      />
    </styled.div>
  </div>
</BrowserRouter>
`;
